### PR TITLE
Enable -Wunused-parameter (GCC, clang)

### PR DESCRIPTION
**Pull request by Nekotekina**
May help to detect some weird bugs.

### CODE REVIEW

File: rpcs3/Emu/Cell/Modules/cellMic.cpp [hunk 1]
```diff
@@ -514,7 +514,7 @@ error_code cellMicInit()
 	return CELL_OK;
 }
 
-error_code cellMicEnd(ppu_thread& ppu)
+error_code cellMicEnd(ppu_thread&)
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/Modules/cellMic.cpp:517]: can't you remove the whole param here and in similar places?

**Comment by Megamouse** [comment]
I saw some random code changes unrelated to the parameters. maybe seperate them into another commit.

File: rpcs3/Emu/Cell/Modules/cellMic.cpp (hunk 1 above)
Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/cellMic.cpp:517]: I don't know, sometimes it can be used as a non-intrusive disambiguator for functions with similar names.

Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/cellMic.cpp:517]: Plus sometimes they are declared in headers as well so I didn't want to change headers.

Comment (Nekotekina) [reply rpcs3/Emu/Cell/Modules/cellMic.cpp:517]: Which leaves literally 3 functions where it can be easily removed, done.

**Comment by Nekotekina** [comment]
Oh no, I'll explode...

**Pull request merged**